src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -994,6 +1034,19 @@ private void AppendHistoryClause(in IndentedStringBuilder.DelimitedScope delimit
             }
         }
 
+        private void AppendMinOrMax(in IndentedStringBuilder.DelimitedScope delimited, SearchOptions context)
+        {
+            delimited.BeginDelimitedElement();
+            if (context.Sort[0].sortOrder == SortOrder.Ascending)
+            {
+                StringBuilder.Append(VLatest.StringSearchParam.IsMin, tableAlias: null).Append(" = 1");
```

Comment:
How did we address the issue where records with no value get returned when a sort is applied? i.e. A Patient that didn't have a name entry?

Reply:
In the SortRewriter class it adds searches for resources that are missing the value. If the sort is in Ascending order then resources missing the value are listed first, if the sort is in Descending order then they are listed last. This is to match SQLs default behavior.